src/box/applier.cc

```diff
@@ -1699,6 +1701,7 @@ applier_apply_tx(struct applier *applier, struct stailq *rows)
 			}
 		}
 	}
+	int rc = 0;
 	applier_synchro_filter_tx(rows);
```

Comment:
AFAIU the problem lies in this `applier_synchro_filter_tx` call.
Let's better make it return an error rather than throw it. This will make `applier_apply_tx` behavior more consistent: it will never throw.